### PR TITLE
Update custom-op repo for manylinux pip packages

### PR DESCRIPTION
**Pull request by yifeif**
- Add option and flag to configure.sh for manylinux2010 packages
- Update build instructions at README.md

### CODE REVIEW

**Comment by yifeif** [comment]
cc: @ewilderj @seanpmorgan

File: README.md [hunk 1]
```diff
@@ -1,18 +1,33 @@
 # TensorFlow Custom Op
 This is a guide for users who want to write custom c++ op for TensorFlow and distribute the op as a pip package. This repository serves as both a working example of the op building and packaging process, as well as a template/starting point for writing your own ops. The way this repository is set up allow you to build your custom ops from TensorFlow's pip package instead of building TensorFlow from scratch. This guarantee that the shared library you build will work with TensorFlow's pip packages.
 
-This guide including example for cpu and gpu ops. 
+This guide including example for both cpu and gpu ops.
+
+Starting from Aug 1, 2019, nightly previews `tf-nightly` and `tf-nightly-gpu`, as well as
+official releases `tensorflow` and `tensorflow-gpu` past version 1.14.1 are now built with a
```
Comment (av8ramit) [inline README.md]: 1.14.0, we did not do a 1.14.1.

Comment (yifeif) [reply README.md]: Done, thanks.

File: README.md [hunk 2]
```diff
@@ -124,24 +139,41 @@ cd my_op
...
+Next, set up a Docker container using the provided Docker image for building and testing the ops. We provide two sets of Docker images for different versions of pip packages. If the pip package you are building against was released before Aug 1, 2019 and has manylinux1 tag, please use Docker images `tensorflow/tensorflow:custom-op` and `tensorflow/tensorflow:custom-op-gpu`, which are based on Ubuntu 14.04. Otherwise, for the newer manylinux2010 packages, please use Docker images `tensorflow/tensorflow:custom-op-ubuntu16` and `tensorflow/tensorflow:custom-op-gpu-ubuntu16` instead. All Docker images come with Bazel pre-installed, as well as the corresponding toolchain used for building the released TensorFlow pacakges. We have seen many cases where dependency version differences and ABI incompatibilities cause the custom op extension users build to not work properly with TensorFlow's released pip packages. Therefore, it is *highly recommended* to use the provided Docker image to build your custom op. To get the CPU Docker image, run one of the following command based on which pip package you are building against:
 ```bash
+# For manylinux1
```
Comment (gunan) [inline README.md]: Not really manylinux1, let's just call this releases before 1.14, or releases labeled manylinux1

File: README.md [hunk 3]
```diff
@@ -124,24 +139,41 @@ cd my_op
...
 docker pull tensorflow/tensorflow:custom-op
```
Comment (gunan) [inline README.md]: Should we push these as `tensorflow/tensorflow:custom-op-ubuntu14` and change them here, preparing for the transition?

File: README.md (hunk 2 above)
Comment (yifeif) [reply README.md]: Updated

File: README.md (hunk 3 above)
Comment (yifeif) [reply README.md]: Sounds good. I pushed new images and updated the guide with the new names.

**Comment by seanpmorgan** [comment]
One last issue. If I try auditwheel repair instead of show:
```
auditwheel repair artifacts/tensorflow_zero_out-0.0.1-cp27-cp27mu-linux_x86_64.whl 
```

It returns `auditwheel: error: cannot find the 'patchelf' tool, which is required`. So we might want to install that in the Dockerfile as well.

**Comment by ewilderj** [comment]
cc @yongtang

**Comment by yifeif** [comment]
Really appreciate the feedback @seanpmorgan!
I have updated the docker images with patchelf and pip, and updated the setup.py according to your comments. Let me know if it's working for you now.
I'm also setting up nightly releases for all of these docker images, so they will automatically get updated with the latest fixes.

**Comment by seanpmorgan** [comment]
Thanks so much @yifeif !

One more fix I discovered after posting:
If you try to compile the GPU op with the installed version of bazel it will error:
`bazel test tensorflow_time_two:time_two_ops_py_test`
```
ERROR: error loading package 'tensorflow_time_two': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
        File "/workspace/gpu/cuda_configure.bzl", line 1297
                _create_local_cuda_repository(repository_ctx)
        File "/workspace/gpu/cuda_configure.bzl", line 1285, in _create_local_cuda_repository
                _tpl(repository_ctx, "crosstool:CROSSTO...", <2 more arguments>)
        File "/workspace/gpu/cuda_configure.bzl", line 1288, in _tpl
                cuda_defines + _get_win_cuda_defines(repository_ctx)
The `+` operator for dicts is deprecated and no longer supported. Please use the `update` method instead. You can temporarily enable the `+` operator by passing the flag --incompatible_disallow_dict_plus=false
```
The fix looks something like this:
https://github.com/tensorflow/addons/pull/389/commits/6b9eb625d2af030cc7db04554c784208827f22f8


Also while we're updating the Dockerfiles we have two more requests which were present in the previous custom-op docker images
1. Install buildifer for bazel linting
```
wget https://github.com/bazelbuild/buildtools/releases/download/0.4.5/buildifier
chmod +x buildifier
mv buildifier /usr/local/bin/.
```
2. Install auditwheel 
```
pip3 install auditwheel
```

**Comment by yifeif** [comment]
Thanks again @seanpmorgan!
I added a patch for fixing `--incompatible_disallow_dict_plus`, as well as for `--incompatible_bzl_disallow_load_after_statement`. 
Also sending another change from internal to add install_buildifier step to the docker image(should mention this PR once it gets into the main repo).

**Comment by yongtang** [comment]
@yifeif Thanks for the PR, looks great!

**Comment by yifeif** [comment]
Thanks all for reviewing!

**Pull request merged**